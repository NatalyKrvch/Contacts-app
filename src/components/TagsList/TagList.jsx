import { Badge } from 'react-bootstrap'

const TagsList = ({ tags }) => {
  return (
    <div>
      {tags.map((tag) => (
        <Badge key={tag} bg="secondary" className="me-2 fw-light">
          {tag}
        </Badge>
      ))}
    </div>
  )
}

export default TagsList
