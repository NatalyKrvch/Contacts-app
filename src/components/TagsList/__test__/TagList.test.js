import { render, screen } from '@testing-library/react';
import TagsList from '../TagList';

describe('TagsList', () => {
  const mockTags = [
    { id: 1, tag: 'Tag1' },
    { id: 2, tag: 'Tag2' },
    { id: 3, tag: 'Tag3' },
  ];

  it('renders tags correctly', () => {
    const { container } = render(<TagsList tags={mockTags} />);
    
    mockTags.forEach(tag => {
      expect(screen.getByTestId(`tag-${tag.id}`)).toBeInTheDocument();
      expect(screen.getByText(tag.tag)).toBeInTheDocument();
    });
    expect(container).toMatchSnapshot();
  });

  it('renders an empty div when no tags are provided', () => {
    const { container } = render(<TagsList tags={[]} />);
    
    expect(screen.getByTestId('tags-list')).toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
});
