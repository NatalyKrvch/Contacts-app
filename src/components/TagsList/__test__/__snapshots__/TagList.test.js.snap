// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsList renders an empty div when no tags are provided 1`] = `
<div>
  <div
    data-testid="tags-list"
  />
</div>
`;

exports[`TagsList renders tags correctly 1`] = `
<div>
  <div
    data-testid="tags-list"
  >
    <span
      class="me-2 fw-light badge bg-secondary"
      data-testid="tag-1"
    >
      Tag1
    </span>
    <span
      class="me-2 fw-light badge bg-secondary"
      data-testid="tag-2"
    >
      Tag2
    </span>
    <span
      class="me-2 fw-light badge bg-secondary"
      data-testid="tag-3"
    >
      Tag3
    </span>
  </div>
</div>
`;
